modules/gateway/peers.go

```diff
@@ -493,7 +493,7 @@ func (g *Gateway) managedConnect(addr modules.NetAddress) error {
 			}
 			defer g.threads.Done()
 
-			err := g.RPC(addr, name, fn)
+			err := g.managedRPC(addr, name, fn)
```

Comment:
I don't think this was a potential deadlock as it was called from a goroutine. But the change fits best practices I think.